### PR TITLE
Manage internal error when trying to delete security relationships without permissions

### PR DESCRIPTION
**Pull request by Espectro123**
|Related issue|
|-|
|[7473](https://github.com/wazuh/wazuh/issues/7473)|

## Description

Hi team,

Trying to delete roles from a user or policies/rules from a role when the user did not have the appropriate permissions was returning an empty list which was causing an uncontrolled exception.

Decorators in the affected functions have been rearranged to avoid this. In addition, the affected endpoints now control if the role or user exists inside the function. This was done previously in the decorator.

The affected endpoints are:

```
DELETE /security/roles/{role_id}/policies
DELETE /security/roles/{role_id}/rules
DELETE /security/users/{user_id}/roles
```
#### framework/wazuh/core/cluster/tests/
```
 55 passed, 4 warnings in 0.40s
```
#### framework/wazuh/core/cluster/dapi/tests/
```
 23 passed, 9 warnings in 0.55s 
```
#### framework/wazuh/core/tests/
```
 654 passed, 9 warnings in 2.20s
```
#### framework/wazuh/tests/
```
 760 passed, 11 warnings in 63.77s (0:01:03)
```
#### framework/wazuh/rbac/tests/
```
 227 passed, 3 warnings in 172.18s (0:02:52)
```
#### api/api/test/
```
 225 passed, 15 warnings in 1.05s
```
## Integration tests related to the issue

```
test_rbac_black_security_endpoints.tavern.yaml 
	 25 passed, 27 warnings

test_rbac_white_security_endpoints.tavern.yaml 
	 25 passed, 27 warnings

test_security_DELETE_endpoints.tavern.yaml 
	 15 passed, 17 warnings

test_security_GET_endpoints.tavern.yaml 
	 18 passed, 21 warnings

test_security_POST_endpoints.tavern.yaml 
	 7 passed, 9 warnings

test_security_PUT_endpoints.tavern.yaml 
	 9 passed, 11 warnings
```

### CODE REVIEW

File: api/test/integration/test_security_DELETE_endpoints.tavern.yaml [hunk 1]
```diff
@@ -137,7 +137,15 @@ stages:
...
       status_code: 200
       json:
-        <<: *error
+        error: 1
+        data:
+          affected_items: []
+          failed_items:
+            - error:
+                code: 4008
+              id: !anything
```
Comment (vicferpoy) [inline api/test/integration/test_security_DELETE_endpoints.tavern.yaml:146]:
```suggestion
              id: 999
```

In addition, please specify in the test name that this policy is a default policy. That is why we get an error 4008 and not a 4007.

**Review by vicferpoy — CHANGES_REQUESTED** [review]
Missing new unit tests for the rest of relationships.

File: framework/wazuh/security.py [hunk 2]
```diff
@@ -828,7 +839,7 @@ def remove_role_rule(role_id, rule_ids):
 
 
 @expose_resources(actions=['security:update'], resources=['role:id:{role_id}', 'policy:id:{policy_ids}'],
-                  post_proc_kwargs={'exclude_codes': [4002, 4007, 4008, 4011]})
+                  post_proc_kwargs={'exclude_codes': [4002, 4007, 4008, 4011,5001]})
```
[the remark this answers is not in the document]
Comment (Espectro123) [reply framework/wazuh/security.py]: It should not be exclude

Fixed

File: api/test/integration/test_security_DELETE_endpoints.tavern.yaml [hunk 3]
```diff
@@ -120,12 +120,12 @@ stages:
...
-      status_code: 200
+      status_code: 404
       json:
-        <<: *error
+        error: 4002
 
-  # DELETE /security/roles/{role_id}/policies/{policy_id}
-  - name: Try to delete a non-existent role-policy (policy non-existent)
+  # DELETE /security/roles/{role_id}/policies/{policy_id}.
+  - name: Try to delete a non-existent role-policy from a default role (policy non-existent)
```
Comment (vicferpoy) [inline api/test/integration/test_security_DELETE_endpoints.tavern.yaml:128]:
```suggestion
  - name: Try to delete a non-existent role-policy (default role, policy non-existent)
```

File: api/test/integration/test_security_DELETE_endpoints.tavern.yaml [hunk 4]
```diff
@@ -137,8 +137,38 @@ stages:
...
       status_code: 200
       json:
-        <<: *error
+        error: 1
+        data:
+          affected_items: []
+          failed_items:
+            - error:
+                code: 4008
+              id: [3]
```
Comment (vicferpoy) [inline api/test/integration/test_security_DELETE_endpoints.tavern.yaml:146]:
```suggestion
              id:
                - 3
```

Consistency.

File: api/test/integration/test_security_DELETE_endpoints.tavern.yaml [hunk 5]
```diff
@@ -137,8 +137,38 @@ stages:
...
+    response:
+      status_code: 200
+      json:
+        error: 1
+        data:
+          affected_items: []
+          failed_items:
+            - error:
+                code: 4007
+              id: [999]
```
Comment (vicferpoy) [inline api/test/integration/test_security_DELETE_endpoints.tavern.yaml:169]:
```suggestion
              id:
                - 999
```

Consistency.

File: framework/wazuh/security.py [hunk 6]
```diff
@@ -3,12 +3,12 @@
 # This program is a free software; you can redistribute it and/or modify it under the terms of GPLv2
 
 import re
+import api.configuration as configuration
+
```
Comment (vicferpoy) [inline framework/wazuh/security.py:7]: PEP8.

**Review by vicferpoy — CHANGES_REQUESTED** [review]

**Review by vicferpoy — APPROVED** [review]
LGTM!

**Review by davidjiglesias — APPROVED** [review]

**Pull request merged**